File: AIS/AIS.BLL/Models/QuestionSet.cs
using System.Collections.Generic;

namespace AIS.BLL.Models
{
    public class QuestionSet
    {
#nullable disable
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<QuestionArea> QuestionAreas { get; set; }
        public IEnumerable<Question> Questions { get; set; }
#nullable enable
        public IEnumerable<int>? QuestionAreaIds { get; set; }
        public IEnumerable<int>? QuestionIds { get; set; }
    }
}

File: AIS.front/ais.front/src/core/interfaces/questionSet.ts
import { Question } from "./question";
import { QuestionArea } from "./questionArea";

export interface QuestionSet{
    id: number,
    name: string,
    questionAreaId: number,
    questionAreas: QuestionArea[] | null,
    questions: Question []
}
export interface QuestionSetAddState {
    name: string,
    questionAreaIds: number[],
    questionIds: number[]
}
